src/Sarif/Readers/AlgorithmKindConverter.cs

```diff
@@ -0,0 +1,97 @@
+        {
+            switch ((AlgorithmKind)value)
+            {
+                case AlgorithmKind.Blake256: { writer.WriteRawValue("\"\"BLAKE-256\"\""); return; }
+                case AlgorithmKind.Blake512: { writer.WriteRawValue("\"BLAKE-512\""); return; }
+                case AlgorithmKind.Ecoh: { writer.WriteRawValue("\"ECOH\""); return; }
+                case AlgorithmKind.Fsb: { writer.WriteRawValue("\"FSB\""); return; }
+                case AlgorithmKind.Gost: { writer.WriteRawValue("\"GOST\""); return; }
+                case AlgorithmKind.Groestl: { writer.WriteRawValue("\"Groestl\""); return; }
+                case AlgorithmKind.Has160: { writer.WriteRawValue("\"HAS-160\""); return; }
+                case AlgorithmKind.Haval: { writer.WriteRawValue("\"HAVAL\""); return; }
+                case AlgorithmKind.JH: { writer.WriteRawValue("\"JH\""); return; }
+                case AlgorithmKind.MD2: { writer.WriteRawValue("\"MD2\""); return; }
+                case AlgorithmKind.MD4: { writer.WriteRawValue("\"MD4\""); return; }
+                case AlgorithmKind.MD5: { writer.WriteRawValue("\"MD5\""); return; }
+                case AlgorithmKind.MD6: { writer.WriteRawValue("\"MD6\""); return; }
+                case AlgorithmKind.RadioGatun: { writer.WriteRawValue("\"RadioGatun\""); return; }
+                case AlgorithmKind.RipeMD: { writer.WriteRawValue("\"RIPEMD\""); return; }
+                case AlgorithmKind.RipeMD128: { writer.WriteRawValue("\"RIPEMD-128\""); return; }
+                case AlgorithmKind.RipeMD160: { writer.WriteRawValue("\"RIPEMD-160\""); return; }
+                case AlgorithmKind.RipeMD320: { writer.WriteRawValue("\"RIPEMD-320\""); return; }
+                case AlgorithmKind.Sha1: { writer.WriteRawValue("\"SHA-1\""); return; }
+                case AlgorithmKind.Sha224: { writer.WriteRawValue("\"SHA-224\""); return; }
+                case AlgorithmKind.Sha256: { writer.WriteRawValue("\"SHA-256\""); return; }
+                case AlgorithmKind.Sha384: { writer.WriteRawValue("\"SHA-384\""); return; }
+                case AlgorithmKind.Sha512: { writer.WriteRawValue("\"SHA-512\""); return; }
+                case AlgorithmKind.Sha3: { writer.WriteRawValue("\"SHA-3\""); return; }
+                case AlgorithmKind.Skein: { writer.WriteRawValue("\"Skein\""); return; }
+                case AlgorithmKind.Snefru: { writer.WriteRawValue("\"Snefru\""); return; }
+                case AlgorithmKind.SpectralHash: { writer.WriteRawValue("\"Spectral Hash\""); return; }
+                case AlgorithmKind.Swifft: { writer.WriteRawValue("\"SWIFFT\""); return; }
+                case AlgorithmKind.Tiger: { writer.WriteRawValue("\"Tiger\""); return; }
+                case AlgorithmKind.Whirlpool: { writer.WriteRawValue("\"Whirlpool\""); return; }
+            }
+            writer.WriteRawValue(@"""unknown""");
```

Comment:
yes. the code as written was actually produced by a regex I applied in VS against enum values I copied from our grammar file. :) I will DRY this out on fixing the other issue you found.